src/Microsoft.OData.Core/UriParser/Binders/MetadataBindingUtils.cs

```diff
@@ -62,19 +62,50 @@ internal static SingleValueNode ConvertToTypeIfNeeded(SingleValueNode source, IE
                 // and the target type is an enum.
                 if (constantNode != null && constantNode.Value != null && (source.TypeReference.IsString() || source.TypeReference.IsIntegral()) && targetTypeReference.IsEnum())
                 {
+                    // String comparison for enum member names is case-sensitive by default.
+                    StringComparison comparison = enableCaseInsensitive ? StringComparison.OrdinalIgnoreCase : StringComparison.Ordinal;
+
                     string memberName = constantNode.Value.ToString();
                     IEdmEnumType enumType = targetTypeReference.Definition as IEdmEnumType;
-                    if (enumType.ContainsMember(memberName, StringComparison.Ordinal))
+
+                    // If the member name is an integral value, we should try to convert it to the enum member name and find the enum member with the matching integral value
+                    if (long.TryParse(memberName, out long memberIntegralValue))

```

Comment:
What should we do if we fail to convert? At the moment we're silently ignoring so someone can be tempted to think the value the provided is valid.

If I do...
```
Customers?$filter=Type eq '3'
```
If it turns out there's no enum member with integral value of 3, and no exception is thrown, I'll just assume there just wasn't a customer with integral value of 3 in my data source. But in actual fact, there's no enum member with an integral value of 3 at all